modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archiver.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'fileutils'
+
+# Built in accordance with the following documentation:
+# https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/platform/practices/zero-silent-failures/remediation.md
+module SimpleFormsApi
+  module S3
+    class SubmissionArchiver < Utils
+      class << self
+        def fetch_presigned_url(benefits_intake_uuid)
+          new(benefits_intake_uuid:).generate_presigned_url
+        end
+
+        def fetch_s3_submission(benefits_intake_uuid)
+          new(benefits_intake_uuid:).download(:submission)
+        end
+
+        def fetch_s3_archive(benefits_intake_uuid)
+          new(benefits_intake_uuid:).download(:archive)
+        end
+      end
+
+      def initialize(parent_dir: 'vff-simple-forms', **options) # rubocop:disable Lint/MissingSuper
+        @parent_dir = parent_dir
+        defaults = default_options.merge(options)
+        @temp_directory_path = build_submission_archive(**defaults)
+        assign_instance_variables(defaults)
+      rescue => e
+        handle_error('SubmissionArchiver initialization failed', e)
+      end
+
+      def upload
+        log_info("Uploading archive: #{benefits_intake_uuid} to S3 bucket")
+
+        upload_temp_folder_to_s3
+        cleanup
+        s3_directory_path
+      rescue => e
+        handle_error("Failed archive upload: #{benefits_intake_uuid}", e)
+      end
+
+      def download(type = :submission)
+        log_info("Downloading #{type}: #{benefits_intake_uuid} to temporary directory: #{temp_directory_path}")
+        send("download_#{type}_from_s3")
+      rescue => e
+        handle_error("Failed #{type} download: #{benefits_intake_uuid}", e)
+      end
+
+      def cleanup
+        FileUtils.rm_rf(temp_directory_path)
+      end
+
+      private
+
+      attr_reader :benefits_intake_uuid, :parent_dir, :submission
+
+      def default_options
+        {
+          attachments: nil,           # The confirmation codes of any attachments which were originally submitted
+          benefits_intake_uuid: nil,  # The UUID returned from the Benefits Intake API upon original submission
+          file_path: nil,             # The local path where the submission PDF is stored
+          metadata: nil,              # Data appended to the original submission headers
+          submission: nil             # The FormSubmission object representing the original data payload submitted
+        }
+      end
+
+      def build_submission_archive(**)
+        SubmissionArchiveBuilder.new(**).run
+      end
+
+      def upload_temp_folder_to_s3
+        Find.find(temp_directory_path) do |path|
```

Comment:
I'm not sure I understand what is going on here. Are we unable to provide the `relative_path` up front? Why do we need to use `Find.find`?

Reply:
Oops, this was probably a carry over from the original script I built this off of. I'll update it. Good catch!

Reply:
Aaaactually, it wasn't a carryover and I just remembered why I did this. This logic iterates through all files within the directory and explicitly reads each one into S3. I went ahead and made a slight change to make it more explicit.